cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -255,16 +255,20 @@ public void execute(int timeoutInSeconds, TxRunnable runnable) throws Transactio
 
       private void executeInternal(Integer timeout, TxRunnable runnable) throws TransactionFailureException {
         int retries = 0;
+        int curTimeout = timeout == null ? -1 : timeout;
         while (true) {
           try {
             if (null == timeout) {
               transactional.execute(runnable);
             } else {
-              transactional.execute(timeout, runnable);
+              transactional.execute(curTimeout, runnable);
             }
             break;
           } catch (TransactionFailureException e) {
             long delay = retryStrategy.nextRetry(e, ++retries);
+            if (timeout != null && retryStrategy instanceof TimeoutRetryOnConflictStrategy) {
+              curTimeout = ((TimeoutRetryOnConflictStrategy) retryStrategy).nextTimeout(e, curTimeout);
```

Comment:
I don't think we should modify the internals of how transactions are executed. This is also a pretty hacky way, checking the instance and doing special logic.

This also doesn't seem correct, we only want to retry failures that were caused by tx timeout. This appears to retry everything.

Increasing the timeout also isn't a pattern that we really want to encourage. It's more like a band-aid until we figure out how best to chunk up the large update we have to do.

Reply:
Removed the change in this class and put all the workaround in the subscriber service